packages/back-end/src/app.ts

```diff
@@ -150,6 +152,26 @@ app.get("/favicon.ico", (req, res) => {
   res.status(404).send("");
 });
 
+let robotsTxt = "";
+app.get("/robots.txt", (_req, res) => {
+  if (!robotsTxt) {
+    const file =
+      process.env.ROBOTS_TXT_PATH || path.join(__dirname, "..", "robots.txt");
```

Comment:
Are you sure that this'll work in production? Files get transpiled and etc it's not clear if this file and the robot.txt file will be placed in the same relative location.

Since the file content is expected to be pretty small, I'd suggest to make the whole file content an env variable. This will also be much easier to override.

Reply:
Hmm, that's a good question. I'd assumed that in prod you'd end up with this code in the `dist` folder the way you do when you build it locally, so stepping out one directory would be good.

I disagree about putting the file contents in an env var personally, as I think the companies that would want to do this setup are more likely to want to version-track their files (incl. robots.txt) & env vars are relatively hard to discover depending on your deployment method.

Reply:
Cool. I don't think env var content should be blocking but making sure that the file is located at the right place should! Also, how would this work in practice when overriding the file location in production for users?

Reply:
So in prod I would assume that a self-hosted org has some build/deploy process to provision their e.g. EC2 instance, and as part of that they'd add a step to put a `robots.txt` somewhere on the machine. Then if they set the `ROBOTS_TXT_PATH` env variable, either globally or as part of the command they use to start their instance, it'll pick that content up the first time the `/robots.txt` url is requested. Does that answer your question?

Reply:
Yeah. For a docker image you would have to do the same but in the `Dockefile` file. 

Not sure how this would allow version tracking but I imagine this would be done on the user side of things?

Definitely something worth documenting!

One more thing: you could also allow both file location AND file content in env variable. As you explained above, file location can be a little more involved and complicated to deploy in production so have a fallback to content env var would be user friendly for those who do not care about tracking the file.

Reply:
Also, using env variable with a default on our end would alleviate any production build concerns.

Reply:
Makes sense, but this is definitely getting out of my wheelhouse, and I don't want to let the perfect be the enemy of the good. I'm gonna merge this as-is now that I added the docs reference & double checked the default path in prod, but I do agree that there's improvements that can be made in future to make this more flexible.

I just want to keep this a lightweight fix for a customer request rather than making it a full-on project/feature that takes longer to go live.